pilot/platform/kube/controller.go

```diff
@@ -326,20 +329,41 @@ func (c *Controller) Instances(hostname string, ports []string,
 					}
 				}
 			}
+			if len(out) == 0 {
+				err := fmt.Errorf("No data found for the query")
```

Comment:
an empty list is not an error. there might be a delay between kubernetes registering a workload, and us making a query, during which the list is empty.